src/services/mongoose/wallet-onchain-pending-receive.ts

```diff
@@ -0,0 +1,111 @@
+import { toSats } from "@domain/bitcoin"
+import { toCents } from "@domain/fiat"
+import { TxStatus } from "@domain/wallets"
+import { WalletCurrency } from "@domain/shared"
+import { CouldNotFindWalletOnChainPendingReceiveError } from "@domain/errors"
+
+import { parseRepositoryError } from "./utils"
+import { WalletOnChainPendingReceive } from "./schema"
+
+export const WalletOnChainPendingReceiveRepository =
+  (): IWalletOnChainPendingReceiveRepository => {
+    const listByWalletId = async ({
```

Comment:
Should we change this to `listByWalletIds` to take a `walletId` array instead? It'll match out ledger service `getTransactionsByWalletIds` method and make it easier to fetch records for multiple wallets like we do in the `getTransactionsForWallets` use-case

Reply:
Updated https://github.com/GaloyMoney/galoy/pull/2622/commits/903ca44ded470fca59c481cbf8fcebccb121948a